flytekit/core/python_function_task.py

```diff
@@ -146,7 +146,9 @@ def compile_into_workflow(
             from flytekit.common.translator import get_serializable
 
             workflow_metadata = WorkflowMetadata(on_failure=WorkflowFailurePolicy.FAIL_IMMEDIATELY)
-            defaults = WorkflowMetadataDefaults(interruptible=False)
+            defaults = WorkflowMetadataDefaults(
+                interruptible=self.metadata.interruptible if self.metadata.interruptible is not None else False
```

Comment:
@wild-endeavor: Can you check this logic?

Reply:
+1